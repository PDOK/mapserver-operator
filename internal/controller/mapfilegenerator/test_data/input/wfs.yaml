metadata:
  labels:
    dataset: dataset
    dataset-owner: datasetOwner
    service-version: v1_0
    theme: theme
spec:
  options:
    automaticCasing: true
    disableWebserviceProxy: false
    includeIngress: false
    prefetchData: false
    rewriteGroupToDataLayers: false
    validateChildStyleNameEqual: false
    validateRequests: false
  service:
    abstract: some "Service" abstract
    accessConstraints: http://creativecommons.org/publicdomain/zero/1.0/deed.nl
    bbox:
      defaultCRS:
        maxx: "1.0"
        maxy: "3.0"
        minx: "0.0"
        miny: "2.0"
    defaultCrs: EPSG:28992
    featureTypes:
    - abstract: feature "1" abstract
      bbox:
        defaultCRS:
          maxx: "1.0"
          maxy: "3.0"
          minx: "0.0"
          miny: "2.0"
      data:
        gpkg:
          blobKey: public/testme/gpkg/file-1.gpkg
          columns:
          - name: featuretype-1-column-1
          - name: featuretype-1-column-2
          geometryType: Point
          tableName: featuretype-1
      datasetMetadataUrl:
        csw:
          metadataIdentifier: datadata-data-data-data-datadatadata
      keywords:
      - featuretype-1-keyword-1
      - featuretype-1-keyword-2
      name: featuretype-1-name
      title: featuretype-1-title
    - abstract: feature "2" abstract
      bbox:
        defaultCRS:
          maxx: "1.0"
          maxy: "3.0"
          minx: "0.0"
          miny: "2.0"
      data:
        postgis:
          columns:
          - alias: alias_featuretype-2-column-1
            name: featuretype-2-column-1
          - name: featuretype-2-column-2
          geometryType: MultiLine
          tableName: featuretype-2
      datasetMetadataUrl:
        csw:
          metadataIdentifier: datadata-data-data-data-datadatadata
      keywords:
      - featuretype-2-keyword-1
      - featuretype-2-keyword-2
      name: featuretype-2-name
      title: featuretype-2-title
    inspire:
      language: ""
      serviceMetadataUrl:
        csw:
          metadataIdentifier: metameta-meta-meta-meta-metametameta
      spatialDatasetIdentifier: ""
    keywords:
    - service-keyword-1
    - service-keyword-2
    - infoFeatureAccessService
    otherCrs:
    - EPSG:28992
    - EPSG:25831
    - EPSG:25832
    - EPSG:3034
    - EPSG:3035
    - EPSG:3857
    - EPSG:4258
    - EPSG:4326
    ownerInfoRef: ""
    prefix: prefix
    title: some Service title
    url: "https://service.pdok.nl/datasetOwner/dataset/theme/wfs/v1_0"
