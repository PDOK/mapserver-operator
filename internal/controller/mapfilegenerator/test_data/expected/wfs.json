{
  "service_title": "some Service title",
  "service_abstract": "some \"Service\" abstract",
  "service_keywords": "service-keyword-1,service-keyword-2,infoFeatureAccessService",
  "service_extent": "0.0 2.0 1.0 3.0",
  "service_wfs_maxfeatures": "1000",
  "service_namespace_prefix": "prefix",
  "service_namespace_uri": "http://prefix.geonovum.nl",
  "service_onlineresource": "https://service.pdok.nl",
  "service_path": "/datasetOwner/dataset/theme/wfs/v1_0",
  "service_metadata_id": "metameta-meta-meta-meta-metametameta",
  "service_accessconstraints": "http://creativecommons.org/publicdomain/zero/1.0/deed.nl",
  "automatic_casing": true,
  "data_epsg": "EPSG:28992",
  "service_debug_level": 0,
  "epsg_list": [
    "EPSG:28992",
    "EPSG:25831",
    "EPSG:25832",
    "EPSG:3034",
    "EPSG:3035",
    "EPSG:3857",
    "EPSG:4258",
    "EPSG:4326"
  ],
  "layers": [
    {
      "name": "featuretype-1-name",
      "title": "featuretype-1-title",
      "abstract": "feature \"1\" abstract",
      "keywords": "featuretype-1-keyword-1,featuretype-1-keyword-2",
      "layer_extent": "0.0 2.0 1.0 3.0",
      "dataset_metadata_id": "datadata-data-data-data-datadatadata",
      "columns": [
        {
          "name": "fuuid"
        },
        {
          "name": "featuretype-1-column-1"
        },
        {
          "name": "featuretype-1-column-2"
        }
      ],
      "geometry_type": "Point",
      "gpkg_path": "/srv/data/gpkg/file-1.gpkg",
      "tablename": "featuretype-1"
    },
    {
      "name": "featuretype-2-name",
      "title": "featuretype-2-title",
      "abstract": "feature \"2\" abstract",
      "keywords": "featuretype-2-keyword-1,featuretype-2-keyword-2",
      "layer_extent": "0.0 2.0 1.0 3.0",
      "dataset_metadata_id": "datadata-data-data-data-datadatadata",
      "columns": [
        {
          "name": "fuuid"
        },
        {
          "name": "featuretype-2-column-1",
          "alias": "alias_featuretype-2-column-1"
        },
        {
          "name": "featuretype-2-column-2"
        }
      ],
      "geometry_type": "MultiLine",
      "tablename": "featuretype-2",
      "postgis": true
    }
  ]
}